Ghost view UI Color change

Vhostym opened the pull request:
Changed InGameUI color to be a burnt orange when in first person ghost view.

Vhostym committed:
Ghost view UI Colorc change

Changed InGameUI color to be a burnt orange when in first person ghost view.

Vhostym commented:
Hi Sabien, can you share the UI mockup images for the first person ghost view? I'm going off of my memory of our Friday meeting and some clearer guidance on what color we want would be helpful.

Vhostym committed:
Added Speed value when in Ghost camera

When in first person Ghost camera you can now see the Speed value in the bottom left of the screen change.

Vhostym committed:
Patch GhostUI coloring

Fixes the color of the filling bar for speed in the UI when the player is spectating as a ghost. This was previously still green.

ambid17 commented on Assets/Editor/ClearLevelCompletion.cs:
```diff
@@ -28,6 +28,7 @@ private void OnGUI()
                 level.levelSaveData.ghostRunPositions = new Vector3[0];
                 level.levelSaveData.ghostRunCameraRotations = new Vector3[0];
                 level.levelSaveData.ghostRunKeys = new KeysPressed[0];
+                level.levelSaveData.ghostRunVelocities = new Vector2[0];
```
nice catch

ambid17 commented on Assets/Scripts/Player/PlayerGhostRun.cs:
```diff
@@ -219,6 +226,7 @@ private void RecordCurrentRunData()
             currentRunPositionData.Add(transform.position);
             currentRunCameraRotationData.Add(playerCamera.transform.eulerAngles);
             currentRunKeyData.Add(GetCurrentKeysPressed());
+            currentRunVelocityData.Add(new Vector2(playerMovement.currentVelocity.x, playerMovement.currentVelocity.z));
```
Youre just using a magnitude later... just saved it off as a float using the magnitude. Save us some data usage

ambid17 committed:
move velocity to float, update ui based off ghost index

Vhostym committed:
Partial Steam Name implementation

LevelData now stores Steam user name if player is connected to SteamClient. Also UIColor is set to work when new white gameUI assets arrive.
Do not merge in this state!

Vhostym committed:
Add new InGameUI assets to prefab

ambid17 committed:
WIP finalizing ghost UI

ambid17 committed:
Ghost UI all working

ambid17 committed:
code cleanup

Vhostym committed:
Final code cleanup and edits
